File: plugin-dotnet-server/src/main/java/jetbrains/buildServer/dotnet/commands/Dnu/RestoreCommandType.java
/*
 * Copyright 2000-2016 JetBrains s.r.o.
 *
 * Licensed under the Apache License, Version 2.0 (the "License").
 * See LICENSE in the project root for license information.
 */

package jetbrains.buildServer.dotnet.commands.Dnu;

import jetbrains.buildServer.dotnet.DnuConstants;
import jetbrains.buildServer.dotnet.commands.CommandType;
import org.jetbrains.annotations.NotNull;

/**
 * Provides parameters for dnu restore command.
 */
public class RestoreCommandType implements CommandType {
    @NotNull
    @Override
    public String getName() {
        return DnuConstants.COMMAND_RESTORE;
    }

    @NotNull
    @Override
    public String getEditPage() {
        return "editRestoreParameters.jsp";
    }

    @NotNull
    @Override
    public String getViewPage() {
        return "viewRestoreParameters.jsp";
    }
}

File: plugin-dotnet-server/src/main/kotlin/jetbrains/buildServer/dotnet/commands/RestoreCommandType.kt
package jetbrains.buildServer.dotnet.commands

import jetbrains.buildServer.dotnet.DotnetCommandType
import jetbrains.buildServer.dotnet.requirements.SDKBasedRequirementFactory

/**
 * Provides parameters for dotnet restore command.
 */
class RestoreCommandType(
    sdkBasedRequirementFactory: SDKBasedRequirementFactory,
) : DotnetCLICommandType(sdkBasedRequirementFactory) {
    override val name: String = DotnetCommandType.Restore.id

    override val editPage: String = "editRestoreParameters.jsp"

    override val viewPage: String = "viewRestoreParameters.jsp"
}